src/test/java/org/broadinstitute/hellbender/utils/svd/SingularValueDecomposerUnitTest.java

```diff
@@ -56,5 +52,23 @@ public static boolean isUnitaryMatrix(final RealMatrix m){
         }
         return true;
     }
+
+    public void assertSVD(final SVD svd, final RealMatrix m) {
+        final RealMatrix U = svd.getU();
+        final RealMatrix V = svd.getV();
+        final double[] s = svd.getSingularValues();
+
+        final RealMatrix S = MatrixUtils.createRealDiagonalMatrix(svd.getSingularValues());
+        final RealMatrix SPrime = MatrixUtils.inverse(S); //it's diagonal so it's ok to just invert
```

Comment:
`S` is only square if `m` is square; otherwise `S` is has the singular values on its diagonal and is padded with zeroes to become rectangular with the right dimensions.  If `m` has dimension q x r then U is q x q, V is r x r, `S` is q x r, and there are min(q,r) singular values.

This test as currently written will break if `m` is not square (hence the more cumbersome code I sketched in the other ticket).  This case needs to be tested e.g. with a random 2 x 3 matrix.